service/controller/v1/resource/drainer/create.go

```diff
@@ -64,6 +57,17 @@ func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
 		return nil
 	}
 
+	// Checking the drained condition must be done after the timeout condition is
+	// managed. This ensures even CRs having a drained condition get the timeout
+	// condition applied. This is to make the provider operators always cleanup
+	// the CRs in case they exist for too long.
+	if drainerConfig.Status.HasDrainedCondition() {
+		r.logger.LogCtx(ctx, "level", "debug", "message", "drainer config status has drained condition")
+		r.logger.LogCtx(ctx, "level", "debug", "message", "canceling resource")
+
+		return nil
+	}
```

Comment:
I don't get why we need to set timeout condition when drained condition is set. When the later is set the node should be removed by the provider operator anyway, correct?

Reply:
Yes this is true. I might have to rethink the cleanup in Azure.